src/stores/coinstore.ts

```diff
@@ -0,0 +1,38 @@
+import { Readable, writable } from "svelte/store";
+import { EXPLORER, ASSETS } from "../constants";
+import { assetHashToCoin, getAssetData } from "../utils/asset";
+import type { Coin } from "../utils/types";
+
+interface State {
+  coins: Coin[]
+}
+
+export type CoinStore = Readable<State> & {
+  updateWithAssets(assets: string[]): Promise<void>
+  getCoin(assetHash: string): Coin
+}
+
+function createCoinStore(initialCoins: Coin[]): CoinStore {
+  let state: State = { coins: initialCoins };
+  const { subscribe, set, update } = writable<State>(state);
+
+  subscribe((v) => state = v);
+
+  return {
+    subscribe,
+    getCoin: (assetHash: string) => state.coins.find(c => c.assetHash === assetHash) || assetHashToCoin(assetHash),
+    updateWithAssets: async (assets: string[]) => {
+      for (const asset of assets) {
+        if (!state.coins.find(c => c.assetHash === asset)) {
+          const newCoin = await getAssetData(asset, EXPLORER);
```

Comment:
what's happen if we got an error here

Reply:
`getCoin` won't find the "Coin" associated to your asset hash. So `assetHashToCoin(assetHash)` (which is basically default values) will be used

Reply:
Note that if the call in getAssetData fails, we also catch the error and return `assetHashToCoin(assetHash)`